tools/testdata-scripts/src/bin/make-testdata.rs

```diff
@@ -55,6 +55,7 @@ fn main() {
     options.insert_feature_gates = true;
     options.use_separate_crates = true;
     options.overwrite = true;
+    options.pretty = true;
```

Comment:
Let's do this: deprecate the `pretty` option and make it a noop. Add a `debug` flag which clearly documents that this is only meant for debugging, slow, and may break if the `format_macro_bodies` doesn't work.

Reply:
I don't think `format_macro_bodies` will stop working; it's more that it may start working _differently_.

Reply:
I'd like clients not to use it mainly because it's slow and the implementation is none of their business...

Reply:
Instability might also worry clients

Reply:
In general I do not see a huge problem in this being a first-class feature. The thing is, yes, it's useful for debugging, but you don't know if you're going to be debugging when you generate the data.

Consider the scenario like that in rustc, where they bake in list formatter data. Most rustc programmers aren't generating this data. It's documented, but you still have to find the documentation and run the steps if you want to regen (and then you need to know these options exist).

It makes a lot of sense for rustc, and other clients, to default to `pretty` just in case someone hacking on the codebase needs to look into here.

So I do not think we should relegate this to an "only meant for debugging" thing, I actually think this should be default behavior.

Reply:
But it can make it look like the data changed when it actually didn't

Reply:
If a random rustc programmer needs to debug our databake output something has gone *seriously* wrong.

Reply:
I really don't think so: they might just be trying to understand what's going on.

Whenever I'm trawling a codebase I'm always annoyed when there's some extremely magical and opaque thing behind the thing I'm trying to understand.

Reply:
It's much faster than data generation for me, it feels like a drop in the bucket.

The other PR, that was somewhat slow (though not hugely so). But yeah we can do that later.